presentation-api/controlling-ua/PresentationAvailability_onchange.html

```diff
@@ -0,0 +1,66 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Monitoring the list of available presentation displays.</title>
+<link rel="author" title="Tomoyuki Shimizu" href="https://github.com/tomoyukilabs">
+<link rel="help" href="http://w3c.github.io/presentation-api/#monitoring-the-list-of-available-presentation-displays">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="common.js"></script>
+
+<p id="notice">The test passes if a "PASS" result appears.<br></p>
+
+<script>
+    // prevent the default timeout
+    setup({explicit_timeout: true});
+
+    var notice = document.getElementById('notice');
+
+    promise_test(function(t) {
+      // clean up the instruction notice when the test ends
+      t.add_cleanup(function() {
+          notice.style.display = 'none';
+      });
+
+      // initialize a presentation request
+      var request = new PresentationRequest(presentationUrls);
+
+      var availability, previousState, eventWatcher;
+
+      // set timeout to observe the presentation availability
+      t.step_timeout(function() {
+        t.force_timeout();
+        t.done();
+      }, 90000);
+
+      var wait = t.step_func(function(a) {
+        availability = a;
+
+        // save the current value of the presentation availability
+        previousState = availability.value;
+
+        // add an instruction notice
+        notice.innerHTML += 'Please make your presentation display '
+                          + (previousState ? 'unavailable' : 'available')
+                          + ' and wait for a minute...';
+
+        // wait until a "change" event is fired
+        if(!eventWatcher)
+          eventWatcher = new EventWatcher(t, availability, ['change', 'change']);
```

Comment:
Hmmm, `change` should not appear twice here. That list represents the list of event types that you want the test harness to listen to, not the sequence of events that you want to check. The sequence of events should rather be given to `wait_for`.

With this code, the test harness will attach two event listeners to the `change` event, which will fire one after the other when a `change` event is received. Now, it will work in our case but that's just "luck": the first one will process the event and resolve the Promise returned by `wait_for`; the second one will try to reject the same Promise (but fail since the Promise has already been resolved) because the event has already been processed. This would break if we were waiting for two change events, as in `wait_for(['change', 'change'])`, before continuing the test.